### DISCUSSION TITLE
last_alexa

### DISCUSSION THREAD

**Discussion by Dregi56** [indomus/forum#1040] [HUB: Home Assistant]
Ho provato a "costruire" un  sensor che mi restituisse il nome dell'ultimo device Alexa con cui ho parlato. Le ho provate tutte...niente da fare, nessuna delle tre modalità che seguono, funziona. ogni tanto c'azzecca... ma molto più spesso persiste il nome del device precedente, o viene sbagliato il nome.
```
- platform: template
    sensors:
      last_alexa:       #l'ultimo device alexa con cui si è parlato
        entity_id:
          - media_player.echoshow
          - media_player.echo_dot_nero
          - media_player.echo_dot_azzurrino
          - media_player.echo_dot_ceruleo
        value_template: >
          {{ states.media_player | selectattr('attributes.last_called','eq',True) | map(attribute='entity_id') | first }}

  - platform: template
    sensors:
      ultimo_alexa:      
        value_template: >
          {{ expand(states.group.echos) | selectattr('attributes.last_called','eq',True) | map(attribute='entity_id') | first }}
  
  - platform: template
    sensors:
      lalexa:
        friendly_name: 'Last Alexa'
        value_template: >-
          {% if is_state_attr('media_player.echoshow','last_called', true) %}
          soggiorno
          {% elif is_state_attr('media_player.echo_dot_nero','last_called', true) %}
          salotto
          {% elif is_state_attr('media_player.echo_dot_azzurrino','last_called', true) %}
          camera girasoli
          {% elif is_state_attr('media_player.echo_dot_ceruleo','last_called', true) %}
          camera matrimoniale
          {% else %}
          da qualche stanza
          {% endif %}
```

**Comment by Dregi56** [comment]
In realtà mi sono accorto che per funzionare funziona, almeno _last_alexa_ il problema è il tempo di aggiornamento. Attual,mente, sto facendo delle prove ie mi sembra che il tempo necessario superi i dieci secondi. Come si può fare in modo che il sensore venga aggiornato ogni secondo?

**Reply by snakuzzo** [reply]
Il tuo template sensor `last_alexa` si aggiorna in base alla valorizzazione del template che avviene ogni volta che vi è l'aggiornamento di una di quelle entità "trattate" in quel sensore...che non è detto si aggiornino tutte nello stesso istante

**Reply by Dregi56** [reply]
Quando parlo ad un device Alexa, questo cambia il suo stato, non  è questo il momento in cui dovrebbe valorizzare il sensore?
Un momento fa lo provavo ed ha _fallato_ ancora. Non sono più certo di nulla, se parlo ad "azzurrino" mi valorizza "echoshow"....mah!
